### PR TITLE
[Modules] New module - Service Endpoint Policy

### PR DESCRIPTION
**Pull request by tyconsulting**
# Description

New module for Service Endpoint policy

[Network - ServiceEndpointPolicies](https://github.com/TY-Consulting/ResourceModules/actions/workflows/ms.network.serviceendpointpolicies.yml)

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

# Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [x] I did format my code

### CODE REVIEW

File: modules/network/service-endpoint-policies/README.md [hunk 1]
```diff
@@ -0,0 +1,186 @@
+# Network ServiceEndpointPolicies `[Microsoft.Network/serviceEndpointPolicies]`
+
+This module deploys Network ServiceEndpointPolicies.
+// TODO: Replace Resource and fill in description
```
Comment (eriqua) [inline modules/network/service-endpoint-policies/README.md:4]: Please remove placeholder
```suggestion
```

File: modules/network/service-endpoint-policies/.test/common/main.test.bicep
Comment (eriqua) [inline modules/network/service-endpoint-policies/.test/common/main.test.bicep:1]: Would you have the bandwidth to also add the min test, containing only the required parameters, in the `.test/min` folder? If not let me know and I'll add an issue to the board to keep track of it.

Comment (tyconsulting) [reply modules/network/service-endpoint-policies/.test/common/main.test.bicep:1]: this is done

File: modules/network/service-endpoint-policies/README.md [hunk 2]
```diff
@@ -0,0 +1,186 @@
+# Network ServiceEndpointPolicies `[Microsoft.Network/serviceEndpointPolicies]`
+
+This module deploys Network ServiceEndpointPolicies.
```
[the remark this answers is not in the document]
Comment (tyconsulting) [reply modules/network/service-endpoint-policies/README.md]: done

**Comment by tyconsulting** [comment]
I've re-ran the pipeline, updated the status badge. hi @eriqua please take another look. thanks

**Comment by tyconsulting** [comment]
Can someone please approve this PR?

File: modules/network/service-endpoint-policies/README.md [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+| `location` | string | `[resourceGroup().location]` | Location for all resources. |
+| `serviceAlias` | string | `''` | The alias indicating if the policy belongs to a service. |
+| `serviceEndpointPolicyDefinitions` | array | `[]` | An Array of service endpoint policy definitions. |
+| `tags` | object | `{object}` | Tags of the resource. |
+
+
+### Parameter Usage: `<ParameterPlaceholder>`
+
+// TODO: Fill in Parameter usage
+
```
[the remark this answers is not in the document]
Comment (eriqua) [reply modules/network/service-endpoint-policies/README.md:42]: In still see the placeholder

File: modules/network/service-endpoint-policies/.test/min/main.test.bicep [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+// Test Execution //
+// ============== //
+
+module testDeployment '../../main.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, location)}-test-${serviceShort}'
+  params: {
+    enableDefaultTelemetry: enableDefaultTelemetry
+    name: '<<namePrefix>>${serviceShort}001'
+    serviceEndpointPolicyDefinitions: []
```
Comment (eriqua) [inline modules/network/service-endpoint-policies/.test/min/main.test.bicep:41]: Please remove optional parameters from min test
```suggestion
```

File: modules/network/service-endpoint-policies/main.bicep [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+  name: name
+  location: location
+  tags: tags
+  properties: {
+    serviceAlias: !empty(serviceAlias) ? serviceAlias : null
+    contextualServiceEndpointPolicies: !empty(contextualServiceEndpointPolicies) ? contextualServiceEndpointPolicies : null
+    serviceEndpointPolicyDefinitions: !empty(serviceEndpointPolicyDefinitions) ? serviceEndpointPolicyDefinitions : null
+  }
+}
+
```
Comment (eriqua) [inline modules/network/service-endpoint-policies/main.bicep:44]: @tyconsulting would you have availability to also add locks and rbac? If not please let us know and I can proceed with the merge and add an issue to track the missing features to be added.

Comment (tyconsulting) [reply modules/network/service-endpoint-policies/main.bicep:55]: @eriqua rbac and locks are added.

File: modules/network/service-endpoint-policies/main.bicep [hunk 6]
```diff
@@ -42,6 +53,28 @@ resource serviceEndpointPolicy 'Microsoft.network/ServiceEndpointPolicies@2022-0
...
+resource serviceEndpointPolicy_lock 'Microsoft.Authorization/locks@2020-05-01' = if (!empty(lock)) {
+  name: '${serviceEndpointPolicy.name}-${lock}-lock'
+  properties: {
+    level: any(lock)
+    notes: lock == 'CanNotDelete' ? 'Cannot delete resource or child resources.' : 'Cannot modify the resource or child resources.'
+  }
+  scope: serviceEndpointPolicy
+}
+
+module serviceEndpointPolicy_roleAssignments '.bicep/nested_roleAssignments.bicep' = [for (roleAssignment, index) in roleAssignments: {
```
Comment (eriqua) [inline modules/network/service-endpoint-policies/main.bicep:65]: @tyconsulting thanks a lot for taking the time to go through all requests. You did an incredible job here! 

As a last thing: could you please add locks and rbac to the common test and regenerate readmes? Then the module would be complete. Thanks!

**Review by eriqua — APPROVED** [review]
🆗

File: modules/network/service-endpoint-policies/main.bicep (hunk 6 above)
Comment (eriqua) [reply modules/network/service-endpoint-policies/main.bicep:65]: Approving for now, missing validation will be integrated in the common test. CC: @ahmadabdalla

**Pull request merged**